webapp/javascript/components/AppSelector/SelectorModal.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import classNames from 'classnames/bind';
+import Input from '@webapp/ui/Input';
+import SelectButton from './SelectButton';
+// eslint-disable-next-line css-modules/no-unused-class
+import styles from './SelectorModal.module.scss';
+
+const cx = classNames.bind(styles);
+
+interface SelectorModalProps {
+  visible: boolean;
+  appNames: string[];
+  selectAppName: (name: string) => void;
+  appName: string;
+}
+
+const DELIMITER = '.';
+
+const getAppNames = (names: string[], name: string) =>
+  names.filter((a) => a.indexOf(name) !== -1);
+
+const SelectorModal = ({
+  visible,
+  appNames,
+  selectAppName,
+  appName,
+}: SelectorModalProps) => {
+  const [filter, setFilter] = useState('');
+  const [selected, select] = useState<string[]>([]);
+
+  const filteredAppNames = useMemo(
+    () =>
+      appNames.filter((n: string) =>
+        n.toLowerCase().includes(filter.trim().toLowerCase())
+      ),
+    [filter, appNames]
+  );
+
```

Comment:
There's a bit of logic here in this component, I would appreciate if you could add comments explaining them in a high level